### PR TITLE
Update expression-syntax.md

### PR DESCRIPTION
**Pull request by ireneontheway**
### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: [pingcap/docs-cn#3313](https://github.com/pingcap/docs-cn/pull/3313)
- Other reference link(s):

### CODE REVIEW

**Comment by ireneontheway** [comment]
@yikeke PTAL

File: expression-syntax.md [hunk 1]
```diff
@@ -1,68 +1,36 @@
 ---
 title: Expression Syntax
 summary: Learn about the expression syntax in TiDB.
-aliases: ['/docs/dev/expression-syntax/','/docs/dev/reference/sql/language-structure/expression-syntax/']
```
Comment (yikeke) [inline expression-syntax.md:4]: The two aliases should not be deleted. Please add them back. Also, [the alias in the Chinese version](https://github.com/pingcap/docs-cn/pull/3313/files#diff-79967a03c6e0a734c3eff924e7a60662L4) need to be added back. Could you help open a PR in docs-cn repo to add it back? @ireneontheway

Comment (ireneontheway) [reply expression-syntax.md]: ok

File: expression-syntax.md [hunk 2]
```diff
@@ -6,63 +6,32 @@ aliases: ['/docs/dev/expression-syntax/','/docs/dev/reference/sql/language-struc
...
-    | Identifier juss stringLit
-    | SubSelect
-    | '!' PrimaryExpression
-    | '~'  PrimaryExpression
-    | '-' PrimaryExpression
-    | '+' PrimaryExpression
-    | "BINARY" PrimaryExpression
-    | PrimaryExpression "COLLATE" StringName
-```
+An expression is a combination of one or more values, operators, or functions. In TiDB, expressions are mainly used in various clauses of the `SELECT` statement, including Group by clause, Where clause, Having clause, Join condition and window function. In addition, some DDL statements will also use expressions, such as the setting of the default values, columns, and partition rules when creating tables.
```
Comment (yikeke) [inline expression-syntax.md:9]:
```suggestion
An expression is a combination of one or more values, operators, or functions. In TiDB, expressions are mainly used in various clauses of the `SELECT` statement, including Group by clause, Where clause, Having clause, Join condition and window function. In addition, some DDL statements also use expressions, such as the setting of the default values, columns, and partition rules when creating tables.
```

File: expression-syntax.md [hunk 3]
```diff
@@ -6,63 +6,32 @@ aliases: ['/docs/dev/expression-syntax/','/docs/dev/reference/sql/language-struc
...
+
+The expressions contain the following types:
```
Comment (yikeke) [inline expression-syntax.md:11]:
```suggestion
The expressions can be divided into the following types:
```

File: expression-syntax.md [hunk 4]
```diff
@@ -6,63 +6,32 @@ aliases: ['/docs/dev/expression-syntax/','/docs/dev/reference/sql/language-struc
...
+
+- Identifier. For reference, see [Schema object names](/schema-object-names.md).
+
+- Predicates, numeric values, strings, date expressions. The [Literal values](/literal-values.md) of these types are also expressions.
+
+- Function calls and window functions. For reference, see [Functions and operators overview](/functions-and-operators/functions-and-operators-overview.md) and [Window functions](/functions-and-operators/window-functions.md)
+
+- ParamMarker (`?`), system variables, user variables and CASE expressions.
+
+The following rules are the expression syntax, which is based on the [parser.y](https://github.com/pingcap/parser/blob/master/parser.y) rules of TiDB parser. For the navigable version of the following syntax diagram, please refer to [TiDB SQL Syntax Diagram](https://pingcap.github.io/sqlgram/#Expression).
```
Comment (yikeke) [inline expression-syntax.md:21]:
```suggestion
The following rules are the expression syntax, which is based on the [parser.y](https://github.com/pingcap/parser/blob/master/parser.y) rules of TiDB parser. For the navigable version of the following syntax diagram, refer to [TiDB SQL Syntax Diagram](https://pingcap.github.io/sqlgram/#Expression).
```
"please" can be dropped when writing our user documents (ref: Google Developer Documentation Style Guide)

**Pull request merged**